packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -378,16 +511,68 @@ const ShippingMultiAddress = ({
                                             </Box>
                                         </VStack>
                                     </Flex>
+
+                                    {/* Add New Address Form - appears inside the product card */}
+                                    {showAddAddressForm[addressKey] && (
+                                        <Box position="relative" mt={4} width="100%">
+                                            <AddressForm
+                                                item={item}
+                                                form={addressForm}
+                                                onSubmit={handleCreateAddress}
+                                                onCancel={() => {
+                                                    setShowAddAddressForm((prev) => ({
+                                                        ...prev,
+                                                        [addressKey]: false
+                                                    }))
+                                                    // Clear form errors using react-hook-form
+                                                    addressForm.clearErrors()
+                                                }}
+                                            />
+                                        </Box>
+                                    )}
                                 </Box>
                             )
                         })}
                     </VStack>
                 </Box>
-                <Box pt={2} w="100%">
-                    <Button type="button" width="full" onClick={onSubmit}>
-                        {formatMessage(submitButtonLabel)}
-                    </Button>
-                </Box>
+                <Button
+                    type="button"
+                    width="full"
+                    mt={2}
+                    isLoading={addressForm.formState.isSubmitting}
+                    {...(isAddressFormOpen && {disabled: true, 'data-disabled': true})}
+                    data-testid="continue-to-shipping-button"
+                    loadingText={formatMessage({
+                        id: 'shipping_multi_address.submit.loading',
+                        defaultMessage: 'Saving address...'
+                    })}
+                    aria-label={formatMessage({
+                        id: 'shipping_multi_address.submit.description',
+                        defaultMessage: 'Continue to next step with selected delivery addresses'
+                    })}
+                    onClick={async () => {
+                        // Prevent click if button is disabled
+                        if (isAddressFormOpen) {
+                            return
+                        }
+
+                        try {
+                            // Now proceed with the checkout
+                            onSubmit()
+                        } catch (error) {
```

Comment:
Try catch to show error here is a bit counter-intuative. the onSubmit here is a prop, yet why does it need to handle the error failing? How about we navigate this job to the parent where we call the api? 

here we simple set the prop with type="submit", we let the form submission flows naturally to the parent without having to assign onClick
```
<Button type=submit>Continue to Shipping</Button>
```

Reply:
I'd like this button to have the same as the "Continue to Shipping Method" of the single address screen for consistent.

Reply:
@shauryemahajanSF Can you please make a note of this? Please implement this change in your WI.

@alexvuong This will be taken care of in the Continue to Shipping Method WI.